### PR TITLE
#19 elementShouldBeVisible Should be public

### PR DESCRIPTION
**Pull request by joegasewicz**
### What

Makes the following parts of the API public:

Describe what you have changed and why.

- `Chrome.ctx` -> `Chrome.Ctx` 
- `UIFeature.waitTimeOut` -> `UIFeature.WaitTimeOut`
- `UIFeature.elementShouldBeVisible` -> `UIFeature.ElementShouldBeVisible`
- `UIFeature.runWithTimeOut` -> `UIFeature.RunWithTimeOut`

So that I can compose custom step functions in a frontend web app, for example:

```
func sectionShouldHaveNthElements(f *componenttest.UIFeature, elementSelector string) func(int) error {
	return func(expectedLength int) error {
		var nodes []*cdp.Node
		err := chromedp.Run(f.Chrome.Ctx,
			f.RunWithTimeOut(&f.Chrome.Ctx, f.WaitTimeOut, chromedp.Tasks{
				chromedp.Nodes(elementSelector, &nodes, chromedp.ByQueryAll),
			}))
		assert.Nil(f, err)
		if err != nil {
			return f.StepError()
		}
		assert.Equal(f, expectedLength, len(nodes))
		return f.StepError()
	}
}

// And call it as follows:
ctx.Step(`^the relations sections should have (\d+) external links$`, sectionShouldHaveNthElements(uiFeature, "a"))
```
Adds `innerListElementsShouldHaveText` function
I want to write a Gherkin step in a frontend web app that can check the breadcrumbs inner text elements exist :
```
 And the page should contain "breadcrumbs" containing list element text "Home, Area" with depth 3
```
This allows the caller to add a testing data attribute to a Dom element e.g. `data-test="breadcrumbs"` & a list of expected text nodes. The function is responsible for querying the node tree up until an nth depth where it iterate over branches as this point.

### How to review
either visually or by pulling down the branch, link to a front end app with 

Describe the steps required to test the changes.
Add this to the mod.go file
```
replace (
	github.com/ONSdigital/dp-component-test => /<PATH_TO_YOUR_LOCAL_REPO>/dp-component-test
)
```

run the tests with 
```
make test-component
```
### Who can review
Anyone

Describe who worked on the changes, so that other people can review.
Myself

### CODE REVIEW

**Review by cookel2 — CHANGES_REQUESTED** [review]

File: ui_feature.go
Comment (cookel2) [inline ui_feature.go:179]: The ctx parameter is not being used
```suggestion
func (f *UIFeature) RunWithTimeOut(timeout time.Duration, tasks chromedp.Tasks) chromedp.ActionFunc {
```

File: ui_feature.go
Comment (cookel2) [inline ui_feature.go:165]: If the RunWithTimeOut signature is changed then this will need to be changed too.

**Comment by cookel2** [comment]
Sorry @joegasewicz I just realised that you asked for this to be reviewed by team 404. So I will approve it after someone from there has done.

**Comment by joegasewicz** [comment]
@cookel2 You can review this should review this as it is agnostic in terms of teams using this code. I've updated the `Who can review` section to `anyone` thank you

File: STEP_DEFINITIONS.md
Comment (cookel2) [reply STEP_DEFINITIONS.md]: The 'What it does'  looks good but the 'Scenario Position' column is still empty.

**Review by cookel2 — CHANGES_REQUESTED** [review]
Just one more little change please.

**Review by cookel2 — APPROVED** [review]
looks good to me

**Pull request merged**
